api/auth_middleware_test.go

```diff
@@ -1,63 +1,101 @@
-	}
+	oauthSA := helpers.CreateRootServiceAccountWithOAuth(t, "oauthUser", "oauth.user@test.com")
+	keyPairSA := helpers.CreateRootServiceAccountWithKeyPair(t, "keyPairUser", "keypair.user@test.com")
+	tokensRepo := helpers.GetRepo(t).Tokens
+	tokens, _ := tokensRepo.FindByEmail(oauthSA.Email)
+	token := tokens[0]
+	emptyMap := make(map[string]string)
 
-	serviceAccountName := response.Header().Get("x-service-account-name")
-	if serviceAccountName != "root" {
-		t.Errorf("Expected service account name %s. Got %s", "root", serviceAccountName)
+	testCases := []struct {
+		testName                string
+		serviceAccount          *models.ServiceAccount
+		requestHeaders          map[string]string
+		expectedResponseHeaders map[string]string
+		expectedResponseCode    int
+	}{
+		{
+			testName:                "KeyPairAuthorization",
+			serviceAccount:          keyPairSA,
+			requestHeaders:          map[string]string{"authorization": fmt.Sprintf("keypair %s:%s", keyPairSA.KeyID, keyPairSA.KeySecret)},
+			expectedResponseHeaders: map[string]string{"x-service-account-name": "keyPairUser"},
+			expectedResponseCode:    http.StatusOK,
+		},
+		{
+			testName:                "OAuthAuthorization",
+			serviceAccount:          oauthSA,
+			requestHeaders:          map[string]string{"authorization": fmt.Sprintf("bearer %s", token.AccessToken)},
+			expectedResponseHeaders: map[string]string{"x-email": oauthSA.Email},
+			expectedResponseCode:    http.StatusOK,
+		},
+		{
+			testName:                "WrongOAuthTokenAuthorization",
+			serviceAccount:          oauthSA,
+			requestHeaders:          map[string]string{"authorization": fmt.Sprintf("bearer %s", "wrong token")},
+			expectedResponseHeaders: emptyMap,
+			expectedResponseCode:    http.StatusUnauthorized,
+		},
+		{
+			testName:                "WrongKeyPairAuthorization",
+			serviceAccount:          keyPairSA,
+			requestHeaders:          map[string]string{"authorization": fmt.Sprintf("keypair %s:%s", "wrong key_id", "wrong_key_secret")},
+			expectedResponseHeaders: emptyMap,
+			expectedResponseCode:    http.StatusUnauthorized,
+		},
+		{
+			testName:                "UndefinedAuthorization",
+			serviceAccount:          nil,
+			requestHeaders:          emptyMap,
+			expectedResponseHeaders: emptyMap,
+			expectedResponseCode:    http.StatusUnauthorized,
+		},
+		{
+			testName:                "UnsupportedAuthorization",
+			serviceAccount:          oauthSA,
+			requestHeaders:          map[string]string{"authorization": fmt.Sprintf("basic %s", b64.StdEncoding.EncodeToString([]byte("user:password")))},
+			expectedResponseHeaders: emptyMap,
+			expectedResponseCode:    http.StatusUnauthorized,
+		},
 	}
-}
 
-func TestAuthMiddlewareBearerShouldAuthenticateUser(t *testing.T) {
-	helpers.CleanupPG(t)
+	for _, testCase := range testCases {
+		t.Run(testCase.testName, func(t *testing.T) {
+			app := helpers.GetApp(t)
+			req, err := http.NewRequest(http.MethodGet, "/service_accounts", nil)
 
```

Comment:
nit: drop the blank line between err's declaration and handling, those should usually stay together.